include/erlang_ls.hrl

```diff
@@ -80,6 +80,11 @@
                      , character := number()
                      }.
 
+%% According to the protocol "If the character value is greater than
+%% the line length it defaults back to the line length.". So we define
+%% a big enough value that represents the end of the line.
+-define(END_OF_LINE, 9999999).
```

Comment:
Why do we need to do this? Can't we just extract the line length from the DB?

Reply:
It's not the line we need but the last colum/character position of the line.

Reply:
I worked around this in haskell-ide-engine (in some other context) by using offset 0 of the next line as the end of the previous one. It is less arbitrary.

Reply:
I tried the `0` approach but it then folded the next line as well, at least this is what I remember when I tried that. I will check again.

Reply:
The problem is in the difference in which the editors are processing the folding range information. 

For example Emacs hides the exact range you provide, if the range is `{StartLine, 0} -> {EndLine, 0}` then it looks like this:
<img width="597" alt="Screenshot 2020-03-09 at 07 21 24" src="https://user-images.githubusercontent.com/1522569/76188075-72cb2400-61d7-11ea-8ac2-eb4e902b6c3d.png">
But in VS Code it looks like this, because apparently it assumes that when the start character is `0` then it should show the whole `StartLine`:
<img width="782" alt="Screenshot 2020-03-09 at 07 13 38" src="https://user-images.githubusercontent.com/1522569/76188219-db1a0580-61d7-11ea-9c09-35e51c94d14f.png">

(Notice that in both cases above the fold is also hiding the new line which shouldn't be part of the range)

When I tried to "solve" the issue by adding one to the start line `{StartLine + 1, 0} -> {EndLine, 0}` it looked like this in Emacs:
<img width="607" alt="Screenshot 2020-03-09 at 07 23 32" src="https://user-images.githubusercontent.com/1522569/76188373-35b36180-61d8-11ea-92cc-90857d60edc9.png">
But like this in VS Code:
<img width="758" alt="Screenshot 2020-03-09 at 07 24 20" src="https://user-images.githubusercontent.com/1522569/76188390-3e0b9c80-61d8-11ea-816c-3b84f8a85dac.png">

The alternative would be to not provide the start and end characters since [according to the protocol](https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/#textDocument_foldingRange) they are both optional and `If not defined, defaults to the length of the end line.`.

Reply:
Can we add the explanation as a comment next to the `99999` value?

Reply:
There is an explanation right above the value.

Reply:
I'm referring to the fact that we're not using a `Line + 1, 0` range because of differences on how clients behave in that case. The reason for having that is that, if we don't explicitly mention it, someone (e.g. an Emacs user) will "fix" that and potentially break things again for VSCode users.

Reply:
Added more detailed comment.